### ISSUE TITLE
About ASIS model

### ISSUE THREAD

**Issue by Obarads** [torch-points3d/torch-points3d#544]
Is there KNNNeighbourFinder for dense ? or will this be add in torch-points3d?
This "dense" means BxNxC input format module such as DenseFPSSampler.

An example of KNNNeighbourFinder for dense is as follows.

```
class DenseKNNNeighbourFinder(BaseNeighbourFinder):
    def __init__(self, k):
        """"
        Parameters
        ----------
        k : int
            Number of Neighbours.
        """
        self.k = k

    def find_neighbours(self, x, y):
        """
        Parameters
        ----------
        x : torch.tensor
            Features of all points. (B,N,C)
        y : torch.tesnor
            Features of center points. (B,M,C)
        
        Returns
        -------
        indexes : torch.tensor
            K neighbours. (B, M, k)
        """
        # indexes = knn_function(x, y, self.k)
        return indexes
```

**Comment by nicolas-chaulet** [comment]
I don't think we have it at the moment, the only thing I could find is this:
https://github.com/nicolas-chaulet/torch-points3d/blob/9966f2350e03165158ba40b9f203aae7a16d31aa/torch_points3d/core/spatial_ops/neighbour_finder.py#L42
You could use that + some reshaping to get a dense version up and running.

**Comment by Obarads** [comment]
Thank you for your reply.
If I want to add DenseKNNNeighbourFinder for a new model PR, could I possibly add it directly to neighbor_finder.py?

Because I thinking of submitting PR of [ASIS](https://openaccess.thecvf.com/content_CVPR_2019/papers/Wang_Associatively_Segmenting_Instances_and_Semantics_in_Point_Clouds_CVPR_2019_paper.pdf) that I am creating with tp3d,  I asked this.

**Comment by nicolas-chaulet** [comment]
That would be amazing! And yes you can just add it there, it's where it belongs!
Best,
Nicolas

**Comment by Obarads** [comment]
I have a question.
Processing of ASIS differs between training and evaluation. 
Input data of ASIS are blocks like torch_geometric.datasets.S3DIS, but   ASIS in evaluation need block data per room because use [BlockMerging](https://github.com/WXinlong/ASIS/blob/d71c3d60e985f5bebe620c8e1a1cb0042fb2f5f6/utils/test_utils.py#L166) (about the same as a function in [SGPN](https://github.com/laughtervv/SGPN/blob/257f0458a5cd734db1642b8805d547dff72f9840/utils/test_utils.py#L165)). On the other hand, ASIS in training use shuffled blocks of all room.

In this case, Should I create scripts/test_segmentation/evaluate.py like scripts/test_registration_scripts/evaluate.py to change processing ?

**Comment by CCInc** [comment]
@Obarads I think the best thing to do in that case is to just make a new s3dis dataset file and if you're in training, use the `torch_geometric` data. Or if in eval/test, use your custom logic.
